text/0000-pub-restricted.md

```diff
@@ -0,0 +1,924 @@
+- Feature Name: pub_restricted
+- Start Date: 2015-12-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Expand the current `pub`/non-`pub` categorization of items with the
+ability to say "make this item visible *solely* to a (named) module
+tree."
+
+The current `crate` is one such tree, and would be expressed via:
+`pub(crate) item`. Other trees can be denoted via a path employed in a
```

Comment:
alternatively `crate` could be allowed at the beginning of paths, thus making this a single syntax